src/simple_cache_client.rs

```diff
@@ -1873,6 +1873,137 @@ impl SimpleCacheClient {
         todo!("This api was reworked and is pending implementation in the rust sdk: https://github.com/momentohq/client-sdk-rust/issues/135");
     }
 
+    /// Fetches a range of elements from a sorted set from a Momento Cache
+    /// selecting by index (rank).
+    ///
+    /// *NOTE*: This is preview functionality and requires that you contact
+    /// Momento Support to enable these APIs for your cache.
+    ///
+    /// # Arguments
+    ///
+    /// * `_cache_name` - name of cache.
+    /// * `_set_name` - name of the set.
+    /// * `_order` - specify ascending or descending order
+    /// * `_limit` - optionally limit the number of results returned
+    /// * `_range` - constrain to a range of elements by index or by score
+    ///
+    /// # Example
+    /// ```
+    /// # fn main() -> momento_test_util::DoctestResult {
+    /// # momento_test_util::doctest(|cache_name, credential_provider| async move {
+    /// use std::time::Duration;
+    /// use momento::SimpleCacheClientBuilder;
+    /// use momento::sorted_set::{Elements, Order};
+    ///
+    /// let mut momento = SimpleCacheClientBuilder::new(credential_provider, Duration::from_secs(30))?
+    ///     .build();
+    ///
+    /// // this will fetch the first 10 elements (with their scores) in the
+    /// // sorted set, sorted by index (rank) in ascending order. Any valid
+    /// // `core::ops::Range` can be used to select and limit the returned
+    /// // elements.
+    /// match momento.sorted_set_fetch_by_index(
+    ///     &cache_name,
+    ///     "test sorted set",
+    ///     Order::Ascending,
+    ///     0..10,
+    ///     true,
+    /// ).await? {
+    ///     Some(elements) => match elements {
+    ///         Elements::ValuesWithScores(elements) => {
+    ///             for element in elements.elements {
```

Comment:
This feels a bit silly, but I think that the only better option is to have the `ValuesWithScores` struct either not exist, or implement the appropriate iterator traits. Alternatively, we can wrap things up with a new type in this crate to provide the iterator.

Reply:
yes, our intent is to provide a concrete Response object type for all APIs, as with Kenny's PR that I linked earlier.

Here's what this response type looks like in the Java SDK for example: https://github.com/momentohq/client-sdk-java/blob/main/momento-sdk/src/main/java/momento/sdk/responses/cache/sortedset/SortedSetFetchResponse.java#L23

not sure if that helps?